packetbeat/protos/mysql/mysql.go

```diff
@@ -650,6 +742,32 @@ func (mysql *mysqlPlugin) receivedMysqlResponse(msg *mysqlMessage) {
 		"error_code":    msg.errorCode,
 		"error_message": msg.errorInfo,
 	})
+	if msg.statementID != 0 {
+		// cache prepare statement response info
+		stmts := mysql.getStmtsMap(msg.tcpTuple.Hashable())
+		if stmts == nil {
+			stmts = mysqlStmtMap{}
+			stmtData := &mysqlStmtData{
+				query:           trans.query,
+				numOfParameters: msg.numberOfParameter,
+			}
+			stmts[msg.statementID] = stmtData
+			mysql.preparestatements.Put(msg.tcpTuple.Hashable(), stmts)
+		} else {
+			if stmts[msg.statementID] == nil {
+				stmtData := &mysqlStmtData{
+					query:           trans.query,
+					numOfParameters: msg.numberOfParameter,
+				}
+				stmts[msg.statementID] = stmtData
+			}
+			mysql.preparestatements.Put(msg.tcpTuple.Hashable(), stmts)
+		}
+		// not publish prepare statement
+		mysql.transactions.Delete(trans.tuple.Hashable())
```

Comment:
I'd argue we want prepare/close statement publishing to be optional :)

Reply:
Can be optional:)